### PR TITLE
Task/6936

### PR DESCRIPTION
**Pull request by GoGoPenguin**
## Description

[redmine issue](https://redmine.kingkit.codes/issues/6936)

## Motivation and Context

- 加入 [vite-plugin-basic-ssl](https://github.com/vitejs/vite-plugin-basic-ssl) 讓開發時也有 ssl，因為要讓 cookies 可以串 OAuth
- 實作 OAuth 登入
- 實作右上角 profile 視窗

## Screenshots (if appropriate)
Untitled

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/hooks/userOAuth.ts
Comment (shengyuan) [inline src/hooks/userOAuth.ts:1]: 建議調整命名為 `useOAuth.ts`，hook 裡的命名，會是 use 開頭，有一致性。🙂

File: src/router/index.ts [hunk 1]
```diff
@@ -48,7 +50,29 @@ const router = createRouter({
...
     },
+    {
+      path: "/:catchAll(.*)",
+      component: LayoutDefault,
+      children: [
+        {
+          path: '',
+          redirect: { name: 'Home' },
+        },
+      ],
```
Comment (shengyuan) [inline src/router/index.ts:61]: 這裡好像可以參考[官網](https://router.vuejs.org/zh/guide/essentials/redirect-and-alias.html#%E9%87%8D%E5%AE%9A%E5%90%91)，直接 `redirect` ？

`const routes = [{ path: '/home', redirect: { name: 'homepage' } }]`

**Review by shengyuan — CHANGES_REQUESTED** [review]

**Review by shengyuan — APPROVED** [review]

File: src/hooks/userOAuth.ts
Comment (GoGoPenguin) [reply src/hooks/userOAuth.ts:1]: 打錯字，已調整

File: src/router/index.ts (hunk 1 above)
Comment (GoGoPenguin) [reply src/router/index.ts]: 已調整

**Review by shengyuan — APPROVED** [review]

**Pull request merged**